src/effects/RealtimeEffectStateUI.cpp

```diff
@@ -0,0 +1,160 @@
+/*  SPDX-License-Identifier: GPL-2.0-or-later */
+/*!********************************************************************
+
+  Audacity: A Digital Audio Editor
+
+  RealtimeEffectStateUI.cpp
+
+  Dmitry Vedenko
+
+**********************************************************************/
+#include "RealtimeEffectStateUI.h"
+
+#include <cassert>
+
+#include "EffectUI.h"
+#include "RealtimeEffectState.h"
+
+#include "EffectManager.h"
+#include "ProjectWindow.h"
+#include "Track.h"
+
+namespace
+{
+const RealtimeEffectState::RegisteredFactory realtimeEffectStateUIFactory { [](auto& state)
+   { return std::make_unique<RealtimeEffectStateUI>(state); }
+};
+} // namespace
+
+
+RealtimeEffectStateUI::RealtimeEffectStateUI(RealtimeEffectState& state)
+    : mRealtimeEffectState(state)
+{
+}
+
+RealtimeEffectStateUI::~RealtimeEffectStateUI()
+{
+   Hide();
+}
+
+bool RealtimeEffectStateUI::IsShown() const noexcept
+{
+   return mEffectUIHost != nullptr;
```

Comment:
Can the non-modal dialog exist but be hidden?  Then this function name might mislead.

Reply:
But, no, I see.